tests/Projection/AbstractPdoEventStoreReadModelProjectorTestCase.php

```diff
@@ -456,27 +458,21 @@ public function it_detects_gap_and_performs_retry(): void
         $projection
             ->fromStream('user')
             ->init(function () {
-                return [];
+                return ['iteration' => 0];
             })
-            ->when([
-                UserCreated::class => function (array $state, Message $event): array {
-                    return $state;
-                },
-                UsernameChanged::class => function (array $state, Message $event): array {
-                    return $state;
-                },
-            ])
-            ->run(false);
-
-        $this->assertEquals(1, $projectionManager->fetchProjectionStreamPositions('test_projection')['user']);
-
-        $this->assertTrue($gapDetection->isRetrying());
+            ->whenAny(
+                function (array $state, Message $event) use ($projectionManager, $gapDetection, $parallelConnection): array {
+                    if ($state['iteration'] === 1) {
+                        Assertion::true($gapDetection->isRetrying());
+                        $parallelConnection->commit();
```

Comment:
I had to change a little bit logic in this test due now each projection will try to finish by itself